src/js/index.js

```diff
@@ -39,3 +39,205 @@ function addTOCInteractivity() {
 $(document).ready(() => {
   addTOCInteractivity();
 });
+
+
+function setupVersionSwitcher() {
+  // Setup Version Switcher
+
+  // Only enable version switcher when window.versionSwitcher is filled by sphinx
+  if (!window.versionSwitcher) {
+    return;
+  }
+  let pageName = window.versionSwitcher.pageName;
+  let versionJsonUrl = window.versionSwitcher.versionJsonUrl;
+  let enableLocaleSupport = window.versionSwitcher.enableLocaleSupport;
+  let allLocales = window.versionSwitcher.allLocales;
+
+  // Remote version should like this.
+  // .name and .alias must be unique in each locale.
+  // It's not necessary to have same versions in all locales.
+  // When locale is enabled, there must be only one .default=true item in each locale to indicate which one should be redirect if target version doesn't exist in target locale.
+
+  /* 
+      let allVersions = {
+        "en": [
+          {
+            "name": "v1.2.0",
+            "url": "v1.2.0",
+            "alias": ["latest"],
+            "default": true
+          },
+          {
+            "name": "v1.1.0",
+            "url": "v1.1.0",
+            "alias": []
+          },
+        ],
+        "zh":[
+          {
+            "name": "v1.0.0",
+            "url": "v1.0.0",
+            "alias": []
+            "default": true
+          },
+        ],
+      };
+      */
+  function parseCurrentURL() {
+    // parseCurrentURL look up current pathname, generate all information about current version and locale.
+
+    let pathname = window.location.pathname;
+
+    // add "index.html" back when browser omit it.
+    if (pageName.endsWith("index.html") && pathname.endsWith("/")) {
+      pathname += "index.html";
+    }
+    if (pathname.slice(-pageName.length) !== pageName) {
+      // Sphinx generated pages should have exactly same suffix
+      throw 'page suffix do not match requirements';
+    }
+
+    // Get base URL by Removing '/' and pageName
+    let baseURL = pathname.slice(0, -(pageName.length + 1));
+    let parts = baseURL.split('/');
+
+    let currentVersion = '';
+    let currentLocale = '';
+
+    if (enableLocaleSupport) {
+      if (parts.length < 1) {
+        throw 'page base URL do not have any locale information';
+      }
+      currentLocale = parts.pop();
+    }
+    if (parts.length < 1) {
+      throw 'page base URL do not have any locale information';
+    }
+    currentVersion = parts.pop();
+    // This is base URL without any version or locate.
+    let globalBaseURL = parts.join('/')
+
+    return {pageName, baseURL, currentVersion, currentLocale, globalBaseURL};
```

Comment:
When I was testing this locally, I was getting an empty string for `currentVersion`, which caused everything downstream to fail for me. I think this may be because the project is serving up the html build directory `/index.html` and not `/en/v1.3/index.html`. I'm not sure if this has to do with Sphinx options when I set up the project or not, but I think this will probably need to be robust against local builds for testing too where the URL may not be fully populated with locale and version.